yjit/src/codegen.rs

```diff
@@ -4023,9 +4036,11 @@ unsafe extern "C" fn build_kwhash(ci: *const rb_callinfo, sp: *const VALUE) -> V
     hash
 }
 
-enum BlockHandler {
+enum SpecVal {
     None,
-    CurrentFrame,
+    BlockISeq(IseqPtr),
+    BlockParamProxy,
+    PrevEP(*const VALUE),
 }
```

Comment:
Would also like a comment explaining what SpecVal is generally used for. In general, it's best not to assume that people are familiar with all the details of the CRuby codebase and send imo.